src/nvim/charset.c

```diff
@@ -1304,6 +1318,18 @@ char_u* skiptowhite_esc(char_u *p) {
   return p;
 }
 
+/// Skip over text until '\n' or NUL.
+///
+/// @param[in]  p  Text to skip over.
+///
+/// @return Pointer to the next '\n' or NUL character.
+char_u *skip_to_newline(const char_u *const p)
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
+  FUNC_ATTR_NONNULL_RET
+{
+  return (char_u *)xstrchrnul((const char *)p, NL);
```

Comment:
seems pretty special-case, why do we need this?

Reply:
Not strictly necessary. Mostly from an old version of the PR (also before I realized we had `xstrchrnul`).
Is nicer to call though, but I can inline if you'd like. :)